### ISSUE TITLE
Sugar: support percentages

### ISSUE THREAD

**Issue by pascallouisperez** [helloeave/worksheets#59]
Support using a `%` sign next to numbers such as `6%` or `1.5%` which would simply interpret the number a `number[n+2]`. From a parsing standpoint, we need to require the `%` to be right next to the number to avoid confusion with `6 % 4` for instance.

**Comment by pascallouisperez** [comment]
We should also consider having a `percentage` type

```
type percentage[n] number[n] constrained_by {
  return 0 <= percentage && percentage <= 1
}

**Comment by workmajj** [comment]
Implemented in #75!
